src/img/utils.c

```diff
@@ -18,7 +18,21 @@ bool pvr_is_alpha(const char *filename)
 		return false;
 	}
 
-	size_t fsize = 512;
+	size_t sector_size = 512;
+	size_t fsize = fs_total(pFile);
+
+	if (fsize <= 0)
+	{
+		dbglog(DBG_INFO, "pvr_is_alpha: empty file");
+		fs_close(pFile);
+		return false;
+	}
+
+	if (fsize > sector_size)
+	{
+		fsize = sector_size;
+	}
+
 	uint8 *data = (uint8 *)memalign(32, fsize);	

```

Comment:
I'm sure you can allocate 512 bytes in the stack for it too. Just don't forget align attribute.
Also you can read just multiple by 32 bytes, so you can reduce size to header size aligned to 32 bytes.

Reply:
Yes, I know, I only added it in case someone modified the GDI and installed an empty PVR.
If there is an empty PVR and the validation does not exist, when using read image, could it flag error or not?

Reply:
For checking empty file you can use FileSize > 0

Reply:
Yes, but in this case I already read the image, it is better with fs_total, right?

Reply:
Yes!

Reply:
So what do I do with this point?

Reply:
Aah, empty you mean transparent? Alpha channel check I see. Just use empty this PVR :) This is rare case.

Reply:
No, empty image really empty, not transparent.
That's what I check in that step and then send whether it's transparent or not.
The transparent image happens to me in several games, that's why I set this condition.

Reply:
Ideally, this should be done simultaneously with the attempt to load the image. But if it is so difficult, leave as is, just use stack instead of memalign.